test/spoof/spoof.go

```diff
@@ -166,71 +166,47 @@ func ResolveEndpoint(ctx context.Context, kubeClientset *kubernetes.Clientset, d
-func (sc *SpoofingClient) Do(req *http.Request) (*Response, error) {
-	// Starting span to capture zipkin trace.
-	traceContext, span := trace.StartSpan(req.Context(), "SpoofingClient-Trace")
-	defer span.End()
-
-	// Check to see if the call to this method is coming from a Poll call.
-	logZipkinTrace := true
-	if req.Header.Get(pollReqHeader) != "" {
-		req.Header.Del(pollReqHeader)
-		logZipkinTrace = false
-	}
-	resp, err := sc.Client.Do(req.WithContext(traceContext))
-	if err != nil {
-		return nil, err
-	}
-
-	defer resp.Body.Close()
-
-	resp.Header.Add(zipkin.ZipkinTraceIDHeader, span.SpanContext().TraceID.String())
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	spoofResp := &Response{
-		Status:     resp.Status,
-		StatusCode: resp.StatusCode,
-		Header:     resp.Header,
-		Body:       body,
-	}
-
-	if logZipkinTrace {
-		sc.logZipkinTrace(spoofResp)
-	}
-
-	return spoofResp, nil
+func (sc *SpoofingClient) Do(req *http.Request, errorRetryCheckers ...ErrorRetryChecker) (*Response, error) {
+	return sc.Poll(req, func(*Response) (bool, error) { return true, nil }, errorRetryCheckers...)
 }
 
 // Poll executes an http request until it satisfies the inState condition or encounters an error.
 func (sc *SpoofingClient) Poll(req *http.Request, inState ResponseChecker, errorRetryCheckers ...ErrorRetryChecker) (*Response, error) {
-	var (
-		resp *Response
-		err  error
-	)
-
-	err = wait.PollImmediate(sc.RequestInterval, sc.RequestTimeout, func() (bool, error) {
-		// As we may do multiple Do calls as part of a single Poll we add this temporary header
-		// to the request to indicate to Do method not to log Zipkin trace, instead it is
-		// handled by this method itself.
-		req.Header.Add(pollReqHeader, "True")
-		resp, err = sc.Do(req)
+	if len(errorRetryCheckers) == 0 {
+		errorRetryCheckers = []ErrorRetryChecker{DefaultErrorRetryChecker}
+	}
+
+	var resp *Response
+	err := wait.PollImmediate(sc.RequestInterval, sc.RequestTimeout, func() (bool, error) {
+		// Starting span to capture zipkin trace.
+		traceContext, span := trace.StartSpan(req.Context(), "SpoofingClient-Trace")
+		defer span.End()
+		rawResp, err := sc.Client.Do(req.WithContext(traceContext))
 		if err != nil {
-			if len(errorRetryCheckers) == 0 {
-				errorRetryCheckers = []ErrorRetryChecker{DefaultErrorRetryChecker}
-			}
 			for _, checker := range errorRetryCheckers {
 				retry, newErr := checker(err)
 				if retry {
 					sc.Logf("Retrying %s: %v", req.URL.String(), newErr)
 					return false, nil
 				}
 			}
+			sc.Logf("NOT Retrying %s: %v", req.URL.String(), err)
 			return true, err
 		}
+		defer rawResp.Body.Close()
 
+		rawResp.Header.Add(zipkin.ZipkinTraceIDHeader, span.SpanContext().TraceID.String())
+		body, err := ioutil.ReadAll(rawResp.Body)
+		if err != nil {
+			return true, err
+		}
+
+		resp = &Response{
```

Comment:
```suggestion
		defer rawResp.Body.Close()
		body, err := ioutil.ReadAll(rawResp.Body)
		if err != nil {
			return true, err
		}

		rawResp.Header.Add(zipkin.ZipkinTraceIDHeader, span.SpanContext().TraceID.String())
		resp = &Response{
```

Reply:
I think I got it, but LMK if I missed anything.